selfdrive/car/ford/radar_interface.py

```diff
@@ -8,27 +8,33 @@
 RADAR_MSGS = list(range(0x120, 0x12F))
 
 
-def _create_radar_can_parser(car_fingerprint):
-  if 'radar' not in DBC[car_fingerprint]:
+def _create_radar_can_parser(CP):
+  if DBC[CP.carFingerprint]['radar'] is None:
     return None
 
-  msg_n = len(RADAR_MSGS)
-  signals = list(zip(['CAN_DET_RANGE'] * msg_n + ['CAN_DET_AZIMUTH'] * msg_n + ['CAN_DET_RANGE_RATE'] * msg_n + ['CAN_DET_VALID_LEVEL'] * msg_n,
-                     RADAR_MSGS * 4,
-                     [0] * msg_n + [0] * msg_n + [0] * msg_n + [0] * msg_n))
-  checks = list(zip(RADAR_MSGS, [20]*msg_n))
+  signals = []
+  checks = []
 
-  return CANParser(DBC[car_fingerprint]['radar'], signals, checks, CANBUS.radar)
+  for addr in RADAR_MSGS:
+    msg = f"MRR_Detection_{addr:03d}"
+    signals += [
+      ("CAN_DET_RANGE", msg),
+      ("CAN_DET_AZIMUTH", msg),
+      ("CAN_DET_RANGE_RATE", msg),
+      ("CAN_DET_VALID_LEVEL", msg),
+    ]
+    checks += [(msg, 20)]
+
+  return CANParser(DBC[CP.carFingerprint]['radar'], signals, checks, CANBUS.radar)
 
 class RadarInterface(RadarInterfaceBase):
   def __init__(self, CP):
     super().__init__(CP)
-    self.validCnt = {key: 0 for key in RADAR_MSGS}
+    self.updated_messages = set()
+    self.trigger_msg = 0x12E
```

Comment:
trigger message cant be 12E because if the radar hadn't got 15 points it'll never trigger, not sure about the best solution for this, need someone on the comma team to weigh in I reckon

Reply:
won't it just send all 64 messages but VALID_LEVEL will be 0?

Reply:
Does the radar continually send all 64 messages? I thought it only sent messages for the points it tracked. I haven't got mine with me otherwise I'd hook it to my can tool and check

Reply:
![Screenshot from 2022-04-23 10-33-36](https://user-images.githubusercontent.com/4038174/164888980-451b87c2-1d78-4f46-b556-6cb314f698d7.png)
I think it sends all or none? It looks like somehow a couple were dropped but almost all of the messages have the same count.